File: hire-event-notifier/app/router/handlers/api/health_check/health_check.py
from fastapi import APIRouter
from .messages.response import HealthCheckResponse
from app.enums.base_api_response_status import BaseAPIResponseStatus

router = APIRouter(tags=["tools"])


@router.get("/health-check", response_model=HealthCheckResponse)
def health_check() -> HealthCheckResponse:
    response = HealthCheckResponse(
        status=BaseAPIResponseStatus.OK,
        msg="Hire Event Notifier service is healthy",
    )

    return response

File: hire-event-processor/internal/app/router/handlers/api/healthcheck.go
package api

import (
	"encoding/json"
	"net/http"
)

type HealthCheckHandler struct{}

type HealthCheckResponse struct {
	Status string `json:"status"`
	Msg    string `json:"msg,omitempty"`
}

func NewHealthCheckHandler() *HealthCheckHandler {
	return &HealthCheckHandler{}
}

func (h *HealthCheckHandler) HealthCheck(w http.ResponseWriter, r *http.Request) {
	response := HealthCheckResponse{
		Status: "ok",
		Msg:    "Hire Event Processor service is healthy",
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	_ = json.NewEncoder(w).Encode(response)
}
